react/ProductName.tsx

```diff
@@ -102,27 +113,42 @@ function ProductName({
     )
   }
 
+  const LinkWrapper =
```

Comment:
Switching between rendering a <Link/> or a `<Fragment/>` is a nice idea! But passing `className` and all of the necessary props for the `Link` even if we're rendering a `<Fragment />` is not a good side effect. 

And I'm pretty sure would also flood the browser's cursor with errors/warnings from React, since `Fragment`s should only receive either a `key` property or none.

Swapping out the `Fragment` for a `div` might be a partial solution, but it isn't great either, since a bunch of props would still be passed to it.

I suggest creating a simple functional component, in this same file, that simply decides what to render in the end, something like:

```js
const LinkWrapper = ({ displayMode, linkProps, className, children }) => {
  if (displayMode === 'plainText') {
    return (
      <Fragment>
        {children}
      </Fragment>
    )
  }

  return (
    <Link className={className} {..linkProps}>
     {children}
    <Link />
  )
}
```

Reply:
I completely agree with the creation of a functional component!